### DISCUSSION TITLE
Upgrade firmware via cli or else, how?

### DISCUSSION THREAD

**Discussion by osoden** [Anime4000/RTL960x#132] [Q&A]
I`ve upgrade v2801f fd511gx-rm0  via web with customized firmware (use https://github.com/Anime4000/RTL960x/tree/main/Tools/emulator) but finaly something wrong
After, have long start,long wait.
 Have web with freezes, try update show 'Upgrade Firmware failed ! (file format error)' - on 192.168.1.1/upgrade2.asp.
On 192.168.1.1/upgrade.asp upload and restart, not any changes.

**Comment by Anime4000** [comment]
the RTL9601CI have a bit issue to flash, try flash normal VF firmware first
however, the emulator is work around ODI stick, check project folder

**Reply by osoden** [reply]
see 100% of CPU
tryed fw`s:
V2801F_V1.9.0-191015.tar
V2801F_V1.9.0-200825.tar

Once i have only telnet, web not worked. 
How i can start boa manually? Maybe upgrade possible with tftp?
Or only pain and suffering?

**Reply by Anime4000** [reply]
copy `boa` to `\tmp`, then chmod +x it and run
